src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -11,38 +12,71 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 import java.io.IOException;
+import java.time.Instant;
+import java.util.UUID;
 
 public class ProfileServletTest {
     private ProfileServlet profileServlet;
     private HttpServletRequest mockRequest;
     private HttpServletResponse mockResponse;
     private RequestDispatcher mockRequestDispatcher;
     private HttpSession mockSession;
+    private UserStore mockUserStore;
+    private User fakeUser;
+    private final String TEST_USERNAME = "test_username";
 
     @Before
     public void setup() throws IOException {
 
         profileServlet = new ProfileServlet();
         mockRequest = Mockito.mock(HttpServletRequest.class);
+        mockSession = Mockito.mock(HttpSession.class);
+
+        setupFakeUser();
+
+        mockUserStore = Mockito.mock(UserStore.class);
+        profileServlet.setUserStore(mockUserStore);
+        Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
```

Comment:
Make sure you are doing your "Mockito.mock"s before your "Mockito.when"s or other functions that use these variables. This just helps make sure that the variable being tested isn't going to change between setup and execution.

Reply:
Done!